discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -40,32 +40,33 @@ def get_metadata(self, multihash, metadata_format, user_replica_set=None):
         """
         logger.warning(f"IPFSCLIENT | get_metadata - {multihash}")
         api_metadata = metadata_format
-        retrieved_from_local_node = False
-        retrieved_from_gateway = False
+        retrieved = False
         start_time = time.time()
 
-        # First try to retrieve from local ipfs node.
-        try:
-            api_metadata = self.get_metadata_from_ipfs_node(multihash, metadata_format)
-            retrieved_from_local_node = api_metadata != metadata_format
-        except Exception:
-            logger.error(
-                f"Failed to retrieve CID from local node, {multihash}", exc_info=True
-            )
-
-        # Else, try to retrieve from gateways.
-        if not retrieved_from_local_node:
-            try:
-                api_metadata = self.get_metadata_from_gateway(
-                    multihash, metadata_format, user_replica_set
-                )
-                retrieved_from_gateway = api_metadata != metadata_format
-            except Exception:
-                logger.error(
-                    f"Failed to retrieve CID from gateway, {multihash}", exc_info=True
-                )
+        with concurrent.futures.ThreadPoolExecutor(max_workers=2) as executor:
+            metadata_futures = []
+            metadata_futures.append(executor.submit(
+                self.get_metadata_from_ipfs_node, multihash, metadata_format
+            ))
+            metadata_futures.append(executor.submit(
+                self.get_metadata_from_gateway, multihash, metadata_format, user_replica_set
+            ))
+            for get_metadata_future in concurrent.futures.as_completed(metadata_futures):
+                original_task = metadata_futures[get_metadata_future]
+                print(f'IPFSCLIENT | original task: {original_task.task}')
+                try:
+                    api_metadata = get_metadata_future.result()
+                    retrieved = api_metadata != metadata_format
+                    break; # use first returned result
+                except Exception as e: # TODO race both, only throw error if both err?
+                    logger.error(
+                        f"ipfs_lib.py | ERROR in metadata_futures parallel processing generated {e}, multihash: {multihash}",
+                        exc_info=True
+                    )
+        retrieved_from_gateway = ''
+        retrieved_from_local_node = ''
 
-        retrieved_metadata = retrieved_from_gateway or retrieved_from_local_node
+        retrieved_metadata = retrieved
```

Comment:
in reference to not using `retrieved` in favor of just `retrieved_metadata`, let's update this too